hummedia/resources.py

```diff
@@ -1,1041 +1,1076 @@
+                }
+            ]
+        }})
+        imgcmd = "avconv -i '%s' -q:v 1 -r 1 -t 00:00:01 -ss 00:00:30 -f image2 '%s'" % (new_file,poster)
+        system(imgcmd.encode('utf-8'))
+        chmod(poster,0775)
+        im=Image.open(poster)
+        im.thumbnail((160,90))
+        im.save(thumb)
+        chmod(thumb,0775)
+
+        if not app.config.get('TESTING'):
+            from gearman import GearmanClient
+            client = GearmanClient(config.GEARMAN_SERVERS)
+            client.submit_job("generate_webm", str(up["id"]))
+        else:
+            from ingest import generate_webm
+            result = generate_webm(file_id=up['id'])
+            if result == "ERROR":
+                raise Exception("Could not convert media file.")
+
+        return (True,)
+    
+    def set_disallowed_atts(self):
+        self.disallowed_atts=["dc:identifier","pid","dc:type","url","ma:duration"]
+        from auth import get_profile
+        atts=get_profile()
+        if not atts['superuser']:
+            self.disallowed_atts.append("dc:creator")
+
+    def update_lastview(self, id, limit=0):
+        '''Update the last view time if the 'view' url parameter is used.'''
+        self.bundle = self.model.find_one({'$or': [{'_id': str(id)}, {'_id': ObjectId(id)}]})
+        if self.bundle:
+            self.bundle = self.auth_filter(self.bundle)
+            if not self.bundle:
+                return action_401()
+            self.bundle[u'@graph'][u'dc:lastviewed'] = datetime.utcnow()
+            self.set_attrs()
+            self.bundle.save()
+            return self.get(id)
+        else:
+            return bundle_400('The ID you submitted is malformed.')
+
+    def get(self,id,limit=0):
+        if self.request.args.get("q") is not None:
+            if limit is 0:
+                return super(MediaAsset, self).get(id, self.max_search_results)
+
+        if "concise" in self.request.args:
+            import json
+            response = super(MediaAsset, self).get(id, limit, ('@graph.pid', '@graph.ma:title','@graph.ma:isMemberOf'))
+            data = json.loads(response.data)
+
+            for i in data:
+              del i['ma:isMemberOf'] # we only want to return the pid and title in concise mode
+
+            response.data = json.dumps(data)
+            return response
+
+        return super(MediaAsset, self).get(id, limit)
+
+    def get_stale(self, date, limit=0):
+        '''Find any videos last watched before the given date, or two years from today if no date given.'''
+        if date:
+            try:
+                datetime.strptime(date, '%Y-%m-%d')
+            except:
+                return bundle_400('The Date you submitted is malformed.')
+        else:
+            default_years = 2
+            today = datetime.utcnow()
+            # Handle leap years.
+            if today.month == 2 and today.day == 29:
```

Comment:
Yes. That seems much cleaner. Seems obvious in hindsight, which probably means it's a good idea.